src/debugger.rs

```diff
@@ -160,24 +161,24 @@ fn build_repl<W: Write + 'static>(output: W) -> DebuggerRepl {
     repl
 }
 
-// Helper function to read a line from stdin
-fn readline(t: f64) -> Result<String, String> {
-    write!(std::io::stdout(), "t={t} $ ").map_err(|e| e.to_string())?;
-    std::io::stdout().flush().map_err(|e| e.to_string())?;
-    let mut buffer = String::new();
-    std::io::stdin()
-        .read_line(&mut buffer)
-        .map_err(|e| e.to_string())?;
-    Ok(buffer)
-}
-
 /// Starts the debugger and pauses execution
 fn start_debugger(context: &mut Context) -> Result<(), IxaError> {
     let t = context.get_current_time();
     let repl = build_repl(std::io::stdout());
     println!("Debugging simulation at t={t}");
+    let mut rl = rustyline::DefaultEditor::new().unwrap();
     loop {
-        let line = readline(t).expect("Error reading input");
+        let line = match rl.readline(&format!("t={t} $ ")) {
+            Ok(line) => line,
+            Err(
+                rustyline::error::ReadlineError::WindowResized
+                | rustyline::error::ReadlineError::Interrupted,
+            ) => continue,
+            Err(rustyline::error::ReadlineError::Eof) => return Ok(()),
+            Err(err) => return Err(IxaError::IxaError(format!("Read error: {err}"))),
+        };
+        rl.add_history_entry(line.clone())
```

Comment:
I think it would be great to maintain history between sessions for the same simulation, but we will need some state in place to make that happen – happy to have that as a followup